### ISSUE TITLE
Feature request: FOR_EACH_BTREE up to 30 in arity

### ISSUE THREAD

**Issue by efferifick** [souffle-lang/souffle#2120]
Hello,

at the moment the Soufflé interpreter limits the arity of b-tree relations to be lower than 20. I would like this to be expanded up to 30. This is a really minor change and I would be happy to submit a pull request. This change would match the maximum arity supported by Provenance. However, I also understand if you prefer to keep the limit to 20.

Thanks!

**Comment by b-scholz** [comment]
There is an issue with testing. If we increase the arity of relations substantially, the compilation time will go up. I suggest having a switch for cmake that limits the arity just for testing so that the release can have a wider range of arity. 

Can you do this?

**Comment by efferifick** [comment]
Yeah, I can work on something like this. Just to make sure I understand, a switch between the default arity and a test arity, where test arity is what is currently in Souffle and the default arity is up to 30? Is that correct?

I am thinking about possible ways to generalize this. Perhaps have cmake generate some .h files with the arity requested instead of switching between two possible arities and then compile the generated files? Would this be better? (I think this is possible, but haven't tried it.)

**Comment by b-scholz** [comment]
Yes - make the default arity 30. Have a compiler switch  (#if ... #endif) so that the compile-time of the interpreter is reduced by lowering the arity (just enough to run the tests). The switch should be controlled in the cmake script and the GitHub actions must be adapted accordingly for continuous testing.
